.github/workflows/main.yml

```diff
@@ -20,57 +20,92 @@ jobs:
             task: r-package
             compiler: gcc
             r_version: 3.6
+            build_type: cmake
           - os: ubuntu-latest
             task: r-package
             compiler: gcc
             r_version: 4.0
+            build_type: cmake
           - os: ubuntu-latest
             task: r-package
             compiler: clang
             r_version: 3.6
+            build_type: cmake
           - os: ubuntu-latest
             task: r-package
             compiler: clang
             r_version: 4.0
+            build_type: cmake
           - os: macOS-latest
             task: r-package
             compiler: gcc
             r_version: 3.6
+            build_type: cmake
           - os: macOS-latest
             task: r-package
             compiler: gcc
             r_version: 4.0
+            build_type: cmake
           - os: macOS-latest
             task: r-package
             compiler: clang
             r_version: 3.6
+            build_type: cmake
           - os: macOS-latest
             task: r-package
             compiler: clang
             r_version: 4.0
+            build_type: cmake
           - os: windows-latest
             task: r-package
             compiler: MINGW
             toolchain: MINGW
             r_version: 3.6
+            build_type: cmake
           - os: windows-latest
             task: r-package
             compiler: MINGW
             toolchain: MSYS
             r_version: 4.0
+            build_type: cmake
           # Visual Studio 2017
           - os: windows-2016
             task: r-package
             compiler: MSVC
             toolchain: MSVC
             r_version: 3.6
+            build_type: cmake
           # Visual Studio 2019
           - os: windows-2019
             task: r-package
             compiler: MSVC
             toolchain: MSVC
             r_version: 4.0
+            build_type: cmake
+          ###############
+          # CRAN builds #
+          ###############
+          - os: windows-latest
+            task: r-package
+            compiler: MINGW
+            toolchain: MSYS
+            r_version: 4.0
+            build_type: cran
+          - os: ubuntu-latest
+            task: r-package
+            compiler: gcc
+            r_version: 4.0
+            build_type: cran
+          - os: macOS-latest
+            task: r-package
+            compiler: gcc
+            r_version: 4.0
+            build_type: cran
     steps:
+      - name: prevent conversion of line endings on Windows
+        if: startsWith(matrix.os, 'windows')
+        shell: pwsh
+        run: git config --global core.autocrlf false
```

Comment:
this is to prevent this issue from `R CMD check`

> checking line endings in shell scripts ... WARNING
Found the following shell script(s) with CR or CRLF line endings:
  configure.ac
Non-Windows OSes require LF line endings.